finesse/gui/temp_control.py

```diff
@@ -0,0 +1,359 @@
+"""Panel and widgets related to temperature monitoring."""
+
+import matplotlib.pyplot as plt
+from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg
+from PySide6.QtCore import Qt
+from PySide6.QtGui import QPixmap
+from PySide6.QtWidgets import (
+    QGridLayout,
+    QGroupBox,
+    QLabel,
+    QLineEdit,
+    QPushButton,
+    QSlider,
+    QSpinBox,
+    QWidget,
+)
+
+# plt.style.use("./finesse_gui.style")
+plt.rcParams["figure.facecolor"] = "black"
+plt.rcParams["axes.facecolor"] = "black"
+plt.rcParams["axes.edgecolor"] = "white"
+plt.rcParams["text.color"] = "white"
+plt.rcParams["xtick.color"] = "white"
+plt.rcParams["ytick.color"] = "white"
+
+
+class BBMonitor(QGroupBox):
+    """Widgets to view the temperature properties."""
+
+    def __init__(self) -> None:
+        """Creates a panel with a graph to monitor the hot and cold bb temperatures."""
+        super().__init__("BB Monitor")
+
+        layout = self._create_controls()
+        self.setLayout(layout)
+
+    def _create_controls(self) -> QGridLayout:
+        """Creates the overall layout for the panel.
+
+        Returns:
+            QGridLayout: The layout containing the figure.
+        """
+        layout = QGridLayout()
+        self._create_figure()
+
+        layout.addWidget(self._canvas, 0, 0)
+
+        return layout
+
+    def _create_figure(self) -> None:
+        """Creates the matplotlib figure to be contained within the panel.
+
+        Returns:
+            None
+        """
+        self._figure, self._ax1 = plt.subplots()
+        self._canvas = FigureCanvasQTAgg(self._figure)
+
+        t = [1040, 1050, 1060, 1070, 1080, 1090, 1100]
+        hot_bb_temp = [55, 57.5, 60, 62.5, 65, 67.5, 70]
+        cold_bb_temp = [1, 1.1, 1.15, 1.2, 1.3, 1.4, 1.5]
+
+        self._bb_hot_line = self._ax1.plot(t, hot_bb_temp, color=[0, 1, 0])
+        self._ax1.set_xlabel("")
+        self._ax1.set_ylabel("HOT BB", color=[0, 1, 0])
+        self._ax1.set_xlim([1045, 1101.4])
+        self._ax1.set_ylim([20, 80])
+
+        self._ax2 = self._ax1.twinx()
+        self._bb_cold_line = self._ax2.plot(t, cold_bb_temp, color=[1, 1, 0])
+        self._ax2.set_ylabel("COLD BB", color=[1, 1, 0])
+        self._ax2.set_ylim([0, 10])
```

Comment:
As with the comment above, we won't want the axis limits and labels to be hard-coded forever, but it's fine to leave them as is for now.